### ISSUE TITLE
Hi  I am confused in using the coco api , especially about segmentation task of mask api. Can anyone give some example about segmentation task relate api please?

### ISSUE THREAD

**Issue by lianshushu** [cocodataset/cocoapi#83]

**Comment by nightrome** [comment]
I assume you are talking about object/thing segmentation. In that case take a look at https://github.com/cocodataset/cocoapi/blob/master/PythonAPI/pycocoDemo.ipynb . For stuff segmentation go to https://github.com/nightrome/cocoapi

**Comment by lianshushu** [comment]
I had read the demon program but did not get help for my work. I need to get the instance segmentaion label for every image. Do you have any example or document about the kind of operation please. Thank you so much!

**Comment by lianshushu** [comment]
when given an image name , I want to get all instances or object label in this image from 0-80 pixel by pixel. The demon code seems to be given an instance class id and get you the image id(name) which made confused.

**Comment by Shanzaay** [comment]
After a lot of effort and search, I was finally able to install coco API. I am trying to run a video captioning code. A neural network for video captioning. These are the libraries which are not executing because anaconda coco API doesn't work anymore. But I installed it anyway. 
**from pycocoevalcap.tokenizer.ptbtokenizer import PTBTokenizer**
**from pycocoevalcap.bleu.bleu import Bleu**
**from pycocoevalcap.meteor.meteor import Meteor**
**from pycocoevalcap.rouge.rouge import Rouge**
**from pycocoevalcap.cider.cider import Cider**

The only line that is executing is this one. 
**from pycocotools.coco import COCO**

How do I run the rest of the above library lines?

Thanks in advance.

**Comment by nightrome** [comment]
@Shanzaay I don't think you properly setup the API. The file is at https://github.com/cocodataset/cocoapi/blob/master/PythonAPI/pycocotools/coco.py . Somehow that folder does not seem to be on your PYTHONPATH.

**Comment by Shanzaay** [comment]
@nightrome it's my first time with this kinda thing. Call me a newbie. Now I am more confused than before. Can you guide me in this? How do I make those libraries execute and how to properly install coco API. 
And yeah you are right now I am having an error even on this line now:  **from pycocotools.coco import COCO**

A step by step guidance will be great. 

Thanks a lot.

**Comment by nightrome** [comment]
@Shanzaay 
The instructions are at: https://github.com/cocodataset/cocoapi
You can verify your PYTHONPATH by doing `echo $PYTHONPATH`.
If there is no entry there for COCO, then check the various other issues in this forum and see if anyone has similar problems.

**Comment by Shanzaay** [comment]
Ok, I'll try it @nightrome. And thanks a bunch.

**Comment by Shanzaay** [comment]
@nightrome I have followed the steps mentioned in the link you share with me. But when I install setup.py I get this error:  

cl : Command line error D8021 : invalid numeric argument '/Wno-cpp'
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\BuildTools\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2

**Comment by Shanzaay** [comment]
@nightrome so here are the steps i performed: 
I clone the repo
i commented this line in setup.py: 
 #extra_compile_args=['-Wno-cpp', '-Wno-unused-function', '-std=c99'],
other wise it give error

once it installed i get something like this:
Using c:\users\myusername\anaconda3\lib\site-packages
Finished processing dependencies for pycocotools==2.0

But i am still having the error. What is it that i am doing wrong?

**Comment by nightrome** [comment]
@Shanzaay Sorry, I haven't tried it on Windows. Please take a look at the various other Windows issues, e.g. https://github.com/cocodataset/cocoapi/issues/9

**Comment by Shanzaay** [comment]
Ok thanks.  @nightrome  does cocoAPI is not supported by Python 3?

**Comment by nightrome** [comment]
Python 3 should not be a problem. But they officially state on the link above that Windows is not supported.

**Comment by virafpatrawala** [comment]
Hey, I have recently written an entire post on [exploring and manipulating the COCO dataset](https://towardsdatascience.com/master-the-coco-dataset-for-semantic-image-segmentation-part-1-of-2-732712631047). Do have a look.

**Comment by virafpatrawala** [comment]
Hey, I have recently written an entire post on [exploring and manipulating the COCO dataset for Image Segmentation](https://towardsdatascience.com/master-the-coco-dataset-for-semantic-image-segmentation-part-1-of-2-732712631047). Do have a look.

**Comment by xml94** [comment]
> Hey, I have recently written an entire post on [exploring and manipulating the COCO dataset for Image Segmentation](https://towardsdatascience.com/master-the-coco-dataset-for-semantic-image-segmentation-part-1-of-2-732712631047). Do have a look.

I read your post, and my got my answer about how to get mask from the dataset. Thanks a lot.
